examples/guides/testing/writing-tests-using-data-providers/README.md

```diff
@@ -0,0 +1,67 @@
+# Hello Service Guide
+
+In this example, we are going to send a message input to a service which will output as an enhanced greeting. 
+
+## Backend
+The backend has been implemented to accept user input and provide an output message. 
+
+## Developement
+It is implemented a service as 'hello' which has the function as sayHello. As an exmaple, if user provides an input as 'John', it will send the response to the client as 'Hello John'. 
+
+## Invoking the Service
+ Navigate to the folder 'sending-a-simple-message-to-a-service' and start the 'hello service.'
+```
+ballerina run hello_service
+```
+The service will be started and you will see the below output. 
+```
+Initiating service(s) in 'hello_service'
+[ballerina/http] started HTTP/WS endpoint 0.0.0.0:9090
+```
+
+Invoke the service as below. 
+```
+curl http://localhost:9090/hello/sayHello -d "John K"
+```
+  
+## Testing
+
+### Data Driven Testing
+
+We can implement data driven tests by providing a function pointer as a data-provider. The function returns a value-set of data and you can iterate the same test over the returned data-set.
+
+In this example we have implemented a simple hello service which gets user input and provide a response. 
+
+We have the main test function 'testHelloServiceResponse' which accepts a single parameter. We have defined our data provider as helloServiceDataProvider. 
+
+```
+@test:Config {
+    dataProvider: "helloServiceDataProvider"
+}
+```
+
+We pass the data to the test cases as below. 
+```
+function helloServiceDataProvider() returns (string[][]) {
+    return [
+    			["John"], 
+    			[" "]
+    	   ];
+}
```

Comment:
The formatting looks odd. Shall we update it by removing extra space characters?

Reply:
Corrected.